src/main/java/org/jvnet/jenkins/plugins/nodelabelparameter/NodeUtil.java

```diff
@@ -15,12 +15,23 @@ private NodeUtil() {}
      * @return <code>true</code> if the job is ok to be used
      */
     public static boolean isNodeOnline(String nodeName) {
-        String controllerLabel = Jenkins.get().getSelfLabel().getName();
+        return isNodeOnline(nodeName, Jenkins.get());
+    }
+
+    /**
+     * Checks whether the given node is available for an execution of the job,
+     *
+     * @param nodeName the name of the node to check
+     * @param jenkins  the Jenkins instance to use
+     * @return <code>true</code> if the job is ok to be used
+     */
+    public static boolean isNodeOnline(String nodeName, Jenkins jenkins) {
+        String controllerLabel = jenkins.getSelfLabel().getName();
```

Comment:
This change is not needed.  JenkinsRule (for JUnit 4) and WithJenkins (for JUnit 5) provide an execution context that has a usable Jenkins main object.  Please use those facilities rather than changing the public signature of the method
```suggestion
        String controllerLabel = Jenkins.get().getSelfLabel().getName();
```